cmd/subaccountsync/main.go

```diff
@@ -1,18 +1,132 @@
 package main
 
 import (
+	"context"
 	"fmt"
 	"log/slog"
 	"os"
-)
+	"os/signal"
+	"time"
+
+	"github.com/kyma-project/kyma-environment-broker/internal/events"
+	"github.com/kyma-project/kyma-environment-broker/internal/storage"
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-project/kyma-environment-broker/internal/kymacustomresource"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 
-const (
-	AppPrefix = "subaccount-sync"
+	kebConfig "github.com/kyma-project/kyma-environment-broker/internal/config"
+	"github.com/vrischmann/envconfig"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	subsync "github.com/kyma-project/kyma-environment-broker/internal/subaccountsync"
 )
 
+const AppPrefix = "subaccount_sync"
+
 func main() {
-	// create slog logger
-	logger := slog.New(slog.NewJSONHandler(os.Stdout, nil))
-	logger.Info(fmt.Sprintf("%s app stub started", AppPrefix))
+	// create context
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer cancel()
+
+	cli := getK8sClient()
+
+	// create and fill config
+	var cfg subsync.Config
+	err := envconfig.InitWithPrefix(&cfg, AppPrefix)
+	if err != nil {
+		fatalOnError(err)
+	}
+
+	logLevel := new(slog.LevelVar)
+	logLevel.Set(cfg.GetLogLevel())
+	slog.SetDefault(slog.New(slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{
+		Level: logLevel,
+	})))
+
+	slog.Info(fmt.Sprintf("Configuration: event window size:%s, event sync interval:%s, accounts sync interval: %s, storage sync interval: %s, queue sleep interval: %s",
+		cfg.EventsWindowSize, cfg.EventsSyncInterval, cfg.AccountsSyncInterval, cfg.StorageSyncInterval, cfg.SyncQueueSleepInterval))
+
+	// create config provider - provider still uses logrus logger
+	configProvider := kebConfig.NewConfigProvider(
+		kebConfig.NewConfigMapReader(ctx, cli, logrus.WithField("service", "storage"), cfg.KymaVersion),
```

Comment:
Slog is used where possible. If some older parts use logrus, a have to provider logrus logger. It will be removed when we migrate our entire code to slog.